config/configLoadHelpers.go

```diff
@@ -108,17 +109,83 @@ func load(r io.Reader, format Format, into any) error {
 	}
 }
 
-func validateConfig(opts *CmdEnv) ([]string, error) {
-	location := opts.ConfigLocation
-	r, format, err := getReaderFor(location)
-	if err != nil {
-		return nil, err
+// This loads all the named configs into destination in the order they are listed.
+// It returns the MD5 hash of the collected configs as a string (if there's only one
+// config, this is the hash of that config; if there are multiple, it's the hash of
+// all of them concatenated together).
+func loadConfigsInto(dest any, locations []string) (string, error) {
+	// start a hash of the configs we read
+	h := md5.New()
+	for _, location := range locations {
+		// trim leading and trailing whitespace just in case
+		location := strings.TrimSpace(location)
+		r, format, err := getReaderFor(location)
+		if err != nil {
+			return "", err
+		}
+		defer r.Close()
+		// write the data to the hash as we read it
+		rdr := io.TeeReader(r, h)
+
+		// when working on a struct, load doesn't overwrite destination values
+		// that are not mentioned in a file, so we can just keep loading into
+		// the same object.
+		if err := load(rdr, format, dest); err != nil {
+			return "", fmt.Errorf("loadConfigsInto unable to load config %s: %w", location, err)
+		}
 	}
-	defer r.Close()
+	hash := hex.EncodeToString(h.Sum(nil))
+	return hash, nil
+}
+
+func loadConfigsIntoMap(dest map[string]any, locations []string) error {
+	for _, location := range locations {
+		// trim leading and trailing whitespace just in case
+		location := strings.TrimSpace(location)
+		r, format, err := getReaderFor(location)
+		if err != nil {
+			return err
+		}
+		defer r.Close()
 
-	var userData map[string]any
-	if err := load(r, format, &userData); err != nil {
-		return nil, fmt.Errorf("validateConfig unable to load config %s: %w", location, err)
+		// when working on a map, load *DOES* overwrite destination values that are not mentioned in a file, so we
```

Comment:
Do you mean if the first config file has `{"d": "15s", "t": "test"}` and the second config file has `{"d": "25s"}`.
The `load()` will overwrite the `"t"` value with empty string after loading these two files?

Reply:
I just rewrote it; what do you think?

Reply:
Thank you! I understand it now